File: ios/Classes/Bridges/TransactionErrorBridge.swift
import Flutter
import SuperwallKit

extension TransactionError {
  func toJson() -> [String: Any] {
    switch self {
      case .pending(let reason):
        return ["type": "pending", "reason": reason]
      case .failure(let message, let product):
        return [
          "type": "failure",
          "message": message,
          "product": product.toJson()
        ]
    }
  }
}

File: android/src/main/kotlin/com/superwall/superwallkit_flutter/bridges/TransactionErrorBridge.kt
package com.superwall.superwallkit_flutter.bridges

import com.superwall.sdk.store.transactions.TransactionError

fun TransactionError.toJson(): Map<String, Any?> {
    return when (this) {
        is TransactionError.Pending -> {
            mapOf("type" to "pending", "reason" to message)
        }
        is TransactionError.Failure -> {
            mapOf("type" to "failure", "message" to message, "product" to product.toJson())
        }
    }
}

